clients/drcachesim/tracer/raw2trace_shared.cpp

```diff
@@ -166,6 +166,31 @@ drmemtrace_get_timestamp_from_offline_trace(const void *trace, size_t trace_size
     return DRMEMTRACE_SUCCESS;
 }
 
+std::string
+read_module_file_bytes(const std::string &modfilename, file_t &modfile,
+                       char *&modfile_bytes)
+{
+    modfile = dr_open_file(modfilename.c_str(), DR_FILE_READ);
```

Comment:
We deliberately isolated all the direct file opens inside raw2trace_directory and we have all i/o in raw2trace or raw2trace_shared indirected via passed-in streams, to better support usage that can't do direct file opens.  So this doesn't seem like it belongs here.

What is raw2trace_directory actually using from raw2trace?  Only check_thread_file?  And check_thread_file pretty much uses just raw2trace_shared code trace_metadata_reader_t::check_entry_thread_start?  Is that a reasonable alternative: move check_thread_file into raw2trace_shared?  That seems to fit better with the split between the modules.

Reply:
Sure, I can try isolating raw2trace_directory from other raw2trace stuff.

Separately:  it might still make sense to take the read_module_file_bytes functionality out of raw2trace_directory. I may just leave an `XXX` comment for now though, because it is convenient to have raw2trace_directory_t own the cleanup of modfile_bytes.

> We deliberately isolated all the direct file opens inside raw2trace_directory and we have all i/o in raw2trace or raw2trace_shared indirected via passed-in streams

Would be nice if we can add some automated check. It is nicer certainly to have that invariant at the module-level.

Reply:
> it might still make sense to take the read_module_file_bytes functionality out of raw2trace_directory.

I mean splitting out read_module_file_bytes from the raw2trace_directory_t class, but still inside the raw2trace_directory module.

Reply:
> What is raw2trace_directory actually using from raw2trace? Only check_thread_file? And check_thread_file pretty much uses just raw2trace_shared code trace_metadata_reader_t::check_entry_thread_start? Is that a reasonable alternative: move check_thread_file into raw2trace_shared? That seems to fit better with the split between the modules.


raw2trace_directory includes lz4/snappy/zlib. They are not needed by analyzers that just need to decode instrs. Would be nicer to not have to link those libs in.

Reply:
Done.